src/main/java/seedu/foodrem/model/item/itemvalidators/ItemPriceValidator.java

```diff
@@ -30,9 +30,10 @@ static Void validateNumericString(String numericString, int maxDecimalPlace, int
                                       String messageNotANumber, String messageTooPrecise, String messageTooLarge,
                                       String messageIsNegative) {
         boolean isParsable = ValidationUtil.isParsableDouble(numericString);
+        boolean hasNoSpecialChars = numericString.matches("[0-9.-]*");
```

Comment:
This regex is weird. Maybe:

```suggestion
        boolean hasNoSpecialChars = numericString.matches("-?[0-9]+.[0-9]*");
```

Reply:
Nevermind, everyting else will be caught by `parseDouble`.